src/core/pointcloud/expression/qgspointcloudexpression.h

```diff
@@ -0,0 +1,263 @@
+    operator QString() const SIP_SKIP;
+
+    /**
+     * Create an empty expression.
+     */
+    QgsPointCloudExpression();
+
+    ~QgsPointCloudExpression();
+
+    /**
+     * Compares two expressions. The operator returns TRUE
+     * if the expression string is equal.
+     */
+    bool operator==( const QgsPointCloudExpression &other ) const;
+
+    /**
+     * Checks if this expression is valid.
+     * A valid expression could be parsed but does not necessarily evaluate properly.
+     */
+    bool isValid() const;
+
+    /**
+     * Returns TRUE if an error occurred when parsing the input expression
+     */
+    bool hasParserError() const;
+
+    /**
+     * Returns parser error
+     */
+    QString parserErrorString() const;
+
+    /**
+     * Returns parser error details including location of error.
+     */
+    QList<QgsExpression::ParserError> parserErrors() const;
+
+    /**
+     * Returns the root node of the expression.
+     *
+     * The root node is NULLPTR if parsing has failed.
+     */
+    const QgsPointCloudExpressionNode *rootNode() const;
+
+    /**
+     * Gets the expression ready for evaluation.
+     * \param block pointer to the QgsPointCloudBlock that will be filtered
+     */
+    bool prepare( const QgsPointCloudBlock *block );
+
+    /**
+     * Gets list of attributes referenced by the expression.
+     */
+    QSet<QString> referencedAttributes() const;
+
+#ifndef SIP_RUN
+
+    /**
+     * Returns a list of all nodes which are used in this expression
+     *
+     * \note not available in Python bindings
+     */
+    QList<const QgsPointCloudExpressionNode *> nodes( ) const;
+
+    /**
+     * Returns a list of all nodes of the given class which are used in this expression
+     *
+     * \note not available in Python bindings
+     */
+    template <class T>
+    QList<const T *> findNodes( ) const
+    {
+      QList<const T *> lst;
+      const QList<const QgsPointCloudExpressionNode *> allNodes( nodes() );
+      for ( const auto &node : allNodes )
+      {
+        const T *n = dynamic_cast<const T *>( node );
+        if ( n )
+          lst << n;
+      }
+      return lst;
+    }
+#endif
+
+    // evaluation
+
+    /**
+     * Evaluate the expression for one point.
+     * \returns 0.0 for false or 1.0 for true.
+     * \param p point number within the block to evaluate
+     */
+    double evaluate( int p );
+
+    /**
+     * Returns TRUE if an error occurred when evaluating last input
+     */
+    bool hasEvalError() const;
+
+    /**
+     * Returns evaluation error
+     */
+    QString evalErrorString() const;
+
+    /**
+     * Sets evaluation error (used internally by evaluation functions)
+     */
+    void setEvalErrorString( const QString &str );
+
+    /**
+     * Set the expression string
+     */
+    void setExpression( const QString &subset );
+
+    /**
+     * Set the expression to a \a QgsExpression.
+     */
+    void setExpression( const QgsExpression &expression );
```

Comment:
I would also remove this setExpression() override and just keep the one accepting QString

Reply:
ok, if we reuse the QgsExpression GUI at some point we may re-introduce it